examples/tokens/ERC4626.vy

```diff
@@ -0,0 +1,257 @@
+# NOTE: Copied from https://github.com/fubuloubu/ERC4626/blob/1a10b051928b11eeaad15d80397ed36603c2a49b/contracts/VyperVault.vy
+from vyper.interfaces import ERC20
+from vyper.interfaces import ERC4626
+
+implements: ERC20
+implements: ERC4626
+
+##### ERC20 #####
+
+totalSupply: public(uint256)
+balanceOf: public(HashMap[address, uint256])
+allowance: public(HashMap[address, HashMap[address, uint256]])
+
+NAME: constant(String[10]) = "Test Vault"
+SYMBOL: constant(String[5]) = "vTEST"
+DECIMALS: constant(uint8) = 18
+
+event Transfer:
+    sender: indexed(address)
+    receiver: indexed(address)
+    amount: uint256
+
+event Approval:
+    owner: indexed(address)
+    spender: indexed(address)
+    allowance: uint256
+
+##### ERC4626 #####
+
+asset: public(ERC20)
+
+event Deposit:
+    depositor: indexed(address)
+    receiver: indexed(address)
+    assets: uint256
+    shares: uint256
+
+event Withdraw:
+    withdrawer: indexed(address)
+    receiver: indexed(address)
+    owner: indexed(address)
+    assets: uint256
+    shares: uint256
+
+
+@external
+def __init__(asset: ERC20):
+    self.asset = asset
+
+
+@view
+@external
+def name() -> String[10]:
+    return NAME
+
+
+@view
+@external
+def symbol() -> String[5]:
+    return SYMBOL
+
+
+@view
+@external
+def decimals() -> uint8:
+    return DECIMALS
+
+
+@external
+def transfer(receiver: address, amount: uint256) -> bool:
+    self.balanceOf[msg.sender] -= amount
+    self.balanceOf[receiver] += amount
+    log Transfer(msg.sender, receiver, amount)
+    return True
+
+
+@external
+def approve(spender: address, amount: uint256) -> bool:
+    self.allowance[msg.sender][spender] = amount
+    log Approval(msg.sender, spender, amount)
+    return True
+
+
+@external
+def transferFrom(sender: address, receiver: address, amount: uint256) -> bool:
+    self.allowance[sender][msg.sender] -= amount
+    self.balanceOf[sender] -= amount
+    self.balanceOf[receiver] += amount
+    log Transfer(sender, receiver, amount)
+    return True
+
+
+@view
+@external
+def totalAssets() -> uint256:
+    return self.asset.balanceOf(self)
+
+
+@view
+@internal
+def _convertToAssets(shareAmount: uint256) -> uint256:
+    totalSupply: uint256 = self.totalSupply
+    if totalSupply == 0:
+        return 0
+
+    # NOTE: `shareAmount = 0` is extremely rare case, not optimizing for it
+    # NOTE: `totalAssets = 0` is extremely rare case, not optimizing for it
+    return shareAmount * self.asset.balanceOf(self) / totalSupply
+
+
+@view
+@external
+def convertToAssets(shareAmount: uint256) -> uint256:
+    return self._convertToAssets(shareAmount)
+
+
+@view
+@internal
+def _convertToShares(assetAmount: uint256) -> uint256:
+    totalSupply: uint256 = self.totalSupply
+    totalAssets: uint256 = self.asset.balanceOf(self)
+    if totalAssets == 0 or totalSupply == 0:
```

Comment:
guessing performance doesn't matter here since it's an example contract, but for fun, if it did, a JUMPI could be shaved off:
```vyper
    if unsafe_mul(totalAssets, convert(totalSupply != 0, uint256)) == 0:
```

Reply:
lol